test/fixtures/incorrect.js

```diff
@@ -174,6 +174,19 @@ const sinon = {};
 
 sinon.useFakeTimers();
 
+// `sort-destructure-keys/sort-destructure-keys`.
+const foobject = {
+  alpha: 1,
+  beta: 2,
+  charlie: 3
```

Comment:
Shouldn't this (and the `noop` lines below) also be out of order? :thinking:

Reply:
that would trigger the `sort-keys`, and I'm not understanding the `noop` part. Could you please provide a bit more context or clarify your question? Sorry and thanks! 😅

Reply:
I am probably the one confused here :) I just assumed that the `incorrect.js` file was meant to demonstrate a non-sorted state in various shapes — an object creation/declaration, an object destructuring, and whatever the `noop` lines are meant to represent (a sequence of function calls perhaps?), with the goal of ensuring (via the tests) that the code indeed sorts these samples into the shapes listed in the `correct.js` file. If my assumptions are indeed misplaced, that's likely what is throwing you off.

Reply:
oh ok no problem 😄 . It seems that the `incorrect.js` contains example of code that will be picked up by the [test function](https://github.com/uphold/eslint-config-uphold/blob/master/test/index.js#L29) and then compared with the array below. 

It seems that the M.O. is to [add a comment with the rule name](https://github.com/uphold/eslint-config-uphold/blob/master/test/fixtures/incorrect.js#L155), then test what [triggers the rule](https://github.com/uphold/eslint-config-uphold/blob/master/test/fixtures/incorrect.js#L160) and use `noop` to [clean-up](https://github.com/uphold/eslint-config-uphold/blob/master/test/fixtures/incorrect.js#L160) so it doesn't "throw" an unused variables warning and "pollutes" that long array of errors/warnings. 

And it also seems that the `correct.js` has a purposely similar structure to that file, maybe to compare almost line-by-line the correct and incorrect code (in a similar style to the eslint.org docs).

creating the `fooobject` with unordered keys would trigger the `sort-keys`, it's [already being tested below](https://github.com/uphold/eslint-config-uphold/blob/master/test/fixtures/incorrect.js#L191) so I felt it was "out-of-scope" so I just sticked to the destructuring part.

Reply:
Ok, thanks for the detailed explanations! It's quite clear now. There's just one thing that's confusing me now, which is why the ``// `sort-destructure-keys/sort-destructure-keys`.`` comment is added before the creation of the object rather than before the destructuring itself. Am I missing something?

Reply:
I've added the comment before all the setup needed for the destructuring as I wanted to keep stuff isolated and not destruct / use an existing object for the example p.e.

I just tried to follow how people did before me (as [in here](https://github.com/uphold/eslint-config-uphold/blob/master/test/fixtures/incorrect.js#L79) p.e.)

Reply:
I see what you mean.

Btw, in that particular case, the entire code under test is within the `class NoDupeClassMembers` block, which is immediately below the comment; a better example of the pattern used in this PR would be [here](https://github.com/uphold/eslint-config-uphold/blob/61478a611758456133fb44804d5fe43cf023d23a/test/fixtures/incorrect.js#L111-L119).

Anyway, since precedence is demonstrated, I suppose this PR is not the place to change the convention (although I still believe it should be done eventually :grin:). Thanks again for indulging my questions!